api/document/js/friendly-error.ts

```diff
@@ -0,0 +1,80 @@
+class XMLErrorInfo {
+  nestedErrors: XMLErrorInfo[];
+  errors: string[];
+  sourceline: number;
+
+  static isXMLError(err: any): boolean {
+    return (typeof err === 'object' && err !== null &&
+            '_sourceline' in err);
+  }
+
+  constructor(err: object) {
+    this.nestedErrors = [];
+    this.errors = [];
+    this.sourceline = parseInt(err['_sourceline'], 10);
```

Comment:
Yeah, I feel like they should, and I'm not sure how/why/where DRF is converting them to strings, because we're setting them as ints in our code. Just filed #975 to address it later.